### PR TITLE
Remove retroactive interface instance syntax

### PR DESCRIPTION
**Pull request by akrmn**
DA-GHC PR: https://github.com/digital-asset/ghc/pull/178
Daml PR: https://github.com/digital-asset/daml/pull/18350

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -515,13 +506,11 @@ renderInterfaceDef InterfaceDef{ifName, ifChoices, ifModule,
       , ifaceDefIface ifName Nothing ifChoices
       , [ "export declare const " <> ifName <> ":"
         , "  damlTypes.InterfaceCompanion<" <> ifName <> ", undefined, '" <> ifaceId <> "'> &"
-        , "  damlTypes.FromTemplate<" <> ifName <> ", " <> retroImplsIntersection <> "> &"
+        , "  damlTypes.FromTemplate<" <> ifName <> ", unknown> &"
```
Comment (paulbrauner-da) [inline language-support/ts/codegen/src/TsCodeGenMain.hs:509]: Does FromTemplate serve any other role than to declare the retroactive instances?

File: compiler/daml-lf-proto-encode/src/DA/Daml/LF/Proto3/EncodeV2.hs [hunk 2]
```diff
@@ -1042,7 +1042,7 @@ encodeDefInterface DefInterface{..} = do
     defInterfaceMethods <- encodeNameMap encodeInterfaceMethod intMethods
     defInterfaceParamInternedStr <- encodeNameId unExprVarName intParam
     defInterfaceChoices <- encodeNameMap encodeTemplateChoice intChoices
-    defInterfaceCoImplements <- encodeNameMap encodeInterfaceCoImplements intCoImplements
+    let defInterfaceCoImplements = mempty
```
[the remark this answers is not in the document]
Comment (akrmn) [reply compiler/daml-lf-proto-encode/src/DA/Daml/LF/Proto3/EncodeV2.hs:1045]: yes, iirc @remyhaemmerle-da wants to clean up the protos as a separate PR

**Review by marciw-da — APPROVED** [review]
LGTM (docs change)

File: compiler/daml-lf-proto-encode/src/DA/Daml/LF/Proto3/EncodeV2.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulbrauner-da) [reply compiler/daml-lf-proto-encode/src/DA/Daml/LF/Proto3/EncodeV2.hs:1045]: Oh, I didn't realize this was a proto, makes sense then.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (paulbrauner-da) [reply language-support/ts/codegen/src/TsCodeGenMain.hs:509]: Oh right, the doc is pretty explicit about it indeed. Thanks!

File: docs/source/daml/code-snippets-dev/Interfaces.daml [hunk 3]
```diff
@@ -107,19 +107,3 @@ template NameOfTemplate
     interface instance YourInterface for NameOfTemplate where
       view = EmptyInterfaceView
 -- EMPTY_INTERFACE_INSTANCE_END
-
```
Comment (paulbrauner-da) [inline docs/source/daml/code-snippets-dev/Interfaces.daml:110]: Don't we need to change the docs as well if that snippet is referenced in the docs?

**Review by paulbrauner-da — APPROVED** [review]
LGTM, thanks!

**Review by paulbrauner-da — APPROVED** [review]
LGTM, thanks!

File: docs/source/daml/code-snippets-dev/Interfaces.daml (hunk 3 above)
Comment (akrmn) [reply docs/source/daml/code-snippets-dev/Interfaces.daml:110]: nope, I used to think this at some point too but `docs.daml.com` points to specific `daml` repo commits, not just `main HEAD`. We **will** need to change the docs _at some point_ but that makes sense for a feature we're removing, it just doesn't have to be now.

**Comment by akrmn** [comment]
https://github.com/digital-asset/ghc/pull/178 has been merged so now we do the GHC_REV dance

**Pull request merged**